internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -332,38 +356,46 @@ func (r *ForeignClusterReconciler) checkPeeringStatus(ctx context.Context,
 		return err
 	}
 
-	var err error
-	foreignCluster.Status.Outgoing.PeeringPhase, err = getPeeringPhase(&outgoingResourceRequestList)
+	status, reason, message, err := getPeeringPhase(&outgoingResourceRequestList)
 	if err != nil {
 		err = fmt.Errorf("[%v] %w in namespace %v", remoteClusterID, err, localNamespace)
 		klog.Error(err)
 		return err
 	}
+	peeringconditionsutils.EnsureStatus(foreignCluster,
+		discoveryv1alpha1.OutgoingPeeringCondition, status, reason, message)
 
-	foreignCluster.Status.Incoming.PeeringPhase, err = getPeeringPhase(&incomingResourceRequestList)
+	status, reason, message, err = getPeeringPhase(&incomingResourceRequestList)
 	if err != nil {
 		err = fmt.Errorf("[%v] %w in namespace %v", remoteClusterID, err, localNamespace)
 		klog.Error(err)
 		return err
 	}
+	peeringconditionsutils.EnsureStatus(foreignCluster,
+		discoveryv1alpha1.IncomingPeeringCondition, status, reason, message)
 	return nil
 }
 
-func getPeeringPhase(resourceRequestList *discoveryv1alpha1.ResourceRequestList) (discoveryv1alpha1.PeeringPhaseType, error) {
+func getPeeringPhase(resourceRequestList *discoveryv1alpha1.ResourceRequestList) (status discoveryv1alpha1.PeeringConditionStatusType,
+	reason, message string, err error) {
 	switch len(resourceRequestList.Items) {
 	case 0:
-		return discoveryv1alpha1.PeeringPhaseNone, nil
+		return discoveryv1alpha1.PeeringConditionStatusNone, noResourceRequestReason,
+			noResourceRequestMessage, nil
 	case 1:
 		resourceRequest := &resourceRequestList.Items[0]
 		desiredDelete := !resourceRequest.Spec.WithdrawalTimestamp.IsZero()
 		deleted := !resourceRequest.Status.OfferWithdrawalTimestamp.IsZero()
 		if desiredDelete && !deleted {
-			return discoveryv1alpha1.PeeringPhaseDisconnecting, nil
+			return discoveryv1alpha1.PeeringConditionStatusDisconnecting, resourceRequestDeletingReason,
+				resourceRequestDeletingMessage, nil
 		}
-		return discoveryv1alpha1.PeeringPhaseEstablished, nil
+		return discoveryv1alpha1.PeeringConditionStatusEstablished, resourceRequestAcceptedReason,
+			resourceRequestAcceptedMessage, nil
```

Comment:
Unclear to me. Is it correct that if a peering has already been deleted, the resulting condition is established?